langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockAnthropicMessageChatModel.java

```diff
@@ -53,47 +69,259 @@ protected Map<String, Object> getRequestParameters(String prompt) {
         parameters.put("anthropic_version", anthropicVersion);
         return parameters;
     }
-    
+
     @Override
     public Response<AiMessage> generate(List<ChatMessage> messages) {
+        return generate(messages, emptyList());
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, ToolSpecification toolSpecification) {
+        return generate(messages, toolSpecification, singletonList(toolSpecification));
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications) {
+        return generate(messages, null, toolSpecifications);
+    }
+
+    private Response<AiMessage> generate(
+            List<ChatMessage> messages,
+            ToolSpecification toolChoiceSpecification,
+            List<ToolSpecification> toolSpecifications
+    ) {
         final String system = getAnthropicSystemPrompt(messages);
-        
+
         List<BedrockAnthropicMessage> formattedMessages = getAnthropicMessages(messages);
-        
+
         Map<String, Object> parameters = getRequestParameters(null);
         parameters.put("messages", formattedMessages);
         parameters.put("system", system);
-        
+
+        if (nonNull(toolChoiceSpecification)) {
+            validateModelIdWithToolsSupport();
+            parameters.put("tool_choice", toAnthropicToolChoice(toolChoiceSpecification));
+        }
+
+        if (!toolSpecifications.isEmpty()) {
+            validateModelIdWithToolsSupport();
+            parameters.put("tools", toAnthropicToolSpecifications(toolSpecifications));
+        }
+
         final String body = Json.toJson(parameters);
-        
+
         InvokeModelResponse invokeModelResponse = withRetry(() -> invoke(body), getMaxRetries());
         final String response = invokeModelResponse.body().asUtf8String();
         BedrockAnthropicMessageChatModelResponse result = Json.fromJson(response, getResponseClassType());
-        
-        return new Response<>(new AiMessage(result.getOutputText()),
+
+        return new Response<>(
+            aiMessageFrom(result),
             result.getTokenUsage(),
-            result.getFinishReason());
+            result.getFinishReason()
+        );
     }
-    
+
+    private Object toAnthropicToolChoice(ToolSpecification toolChoiceSpecification) {
+        ObjectNode toolChoiceNode = new ObjectMapper().createObjectNode();
+        toolChoiceNode.put("type", "tool");
+        toolChoiceNode.put("name", toolChoiceSpecification.name());
+        return toolChoiceNode;
+    }
+
+    private void validateModelIdWithToolsSupport() {
```

Comment:
Could you please write a unit test for this method and test various versions?

Reply:
I refactored the test `testFunctionCallingWithBedrockAnthropicChatModelWithoutToolsSupport` to support the list of models.

Reply:
Oh, his test is now hard to read and understand :( Also it seems there are copy-paste errors (model names).

What I meant is a **unit** test for this specific method. It can be made package-private and static and accept `String model`, then it will be very easy to test in isolation.

Reply:
> Oh, his test is now hard to read and understand :( Also it seems there are copy-paste errors (model names).
> 
> What I meant is a **unit** test for this specific method. It can be made package-private and static and accept `String model`, then it will be very easy to test in isolation.

![image](https://github.com/user-attachments/assets/5a77cad2-2796-4acf-87df-10d64aae03a7)